### PR TITLE
Refactors DetailsList variants to their own sub-page on fabric-website

### PR DESCRIPTION
**Pull request by KevinTCoughlin**
# Test URL

http://53b033a3.ngrok.io/#/components/detailslist

#### Pull request checklist

- [x] Include a change request file using `$ npm run change`

#### Description of changes

@aditima suggested that we begin breaking-out component variants into their own sub-nav page. Some of the reasons for this change include:

- Improved perceived performance with less components rendering per page. `DetailsList` benefiting the most.
- Improved UX re: finding the variant per component needed via sub-nav instead of scrolling. The change also makes variants URL addressable.
- Organization as the repo scales for contributors

I spoke with @Jahnp who has some other work on the site ongoing, so my goal here would be to improve the current site a bit without negatively impacting that team's work. We already have support for child pages via:

https://github.com/OfficeDev/office-ui-fabric-react/blob/97c9717fee948a383084ab0caedc66b0d5a441bd/apps/fabric-website/src/components/Nav/Nav.types.ts#L44

So this change starts with `DetailsList` as an MVP with the rest to come if we agree on the approach. Each variant is broken out to its own page with the existing page skeleton surrounding it.

#### Requires Discussion

There are a few open questions which require owner / design feedback:

- [x] Do we want to render the "Overview" section per variant? If so what text should we display there?

> The decision here was for sub-pages to only render the example component.

- [x] Do we want to render the "Variants" header per sub-section which is a "Variant"? That seems redundant.

> The decision here was to change the text to be the example's header if only 1 example on the page, otherwise "Variants".

- [x] Do we want to render the API references per sub-section or link to it as part of the root page or its own page?

> Only render on root component page.

- [x] Feedback re: URL paths per "Variant" sub-section

> Fine, just remove hyphens for consistency.

- [x] Feedback re: Page titles per "Variant" sub-section

> Fine as-is unless anyone comments otherwise.

- [x] Feedback re: Nav element text / titles per "Variant" sub-section

> Fine as-is unless anyone comments otherwise.

- [x] Do I only refactor `DetailsList` or would we want all components refactored to sub-pages at once?

> Only `DetailsList` for now as its in dire need of splitting.

Here's a screenshot demonstrating the issues mentioned above that require resolution:

ss

#### Focus areas to test

- Clone this branch
- Install / Build the repo
- Navigate to the `fabric-website` app and run `npm start`
- Navigate to `DetailsList` and the sub-sections this branch adds

I've tagged a few folks that have been mentioned as having an opinion on this change. Please feel free to tag more so that we can reach a consensus 😄

### CODE REVIEW

File: apps/fabric-website/src/components/App/AppState.tsx [hunk 1]
```diff
@@ -199,7 +199,63 @@ export const AppState: IAppState = {
...
+            },
+            {
+              title: 'Grouped',
+              url: '#/components/detailslist/grouped',
+              component: () => <LoadingComponent title='DetailsList - Grouped' />,
+              getComponent: cb => require.ensure([], (require) => cb(require<any>('../../pages/Components/DetailsList/DetailsListGroupedComponentPage').DetailsListGroupedComponentPage))
+            },
+            {
+              title: 'Custom Item Columns',
+              url: '#/components/detailslist/custom-item-columns',
```
Comment (lynamemi) [inline apps/fabric-website/src/components/App/AppState.tsx]: For consistency, we haven't been using hyphens for other examples.

**Comment by KevinTCoughlin** [comment]
The **office-ui-fabric-react** local dev pages need to be updated to support these sub-examples now that they are moved from the main page. The nav mechanism is different between it and the fabric website proper.

**Comment by micahgodbolt** [comment]
Is there an issue/PR already created for the dev pages updates?

**Comment by micahgodbolt** [comment]
@KevinTCoughlin is there a timeframe for the local dev page updates? Can we close this PR until you're ready for it to be merged?

**Comment by Jahnp** [comment]
Just FYI that Kevin's out on paternity leave for a while. I'd look to @aditima if we want to keep this open or have it picked up by someone else until he returns.

**Comment by micahgodbolt** [comment]
I was just reminded that we can always close the PRs down without deleting them. Conversations can still happen, it just won't show up in the PR list.

**Comment by dzearing** [comment]
Closing until Kevin is back and can address the rest.

**Pull request closed without merging**